diff --git a/src/vulkan/gpu_tex.c b/src/vulkan/gpu_tex.c
index a419ffdc..19249e19 100644
--- a/src/vulkan/gpu_tex.c
+++ b/src/vulkan/gpu_tex.c
@@ -1144,10 +1144,15 @@ skip_blocking:
 
 pl_tex pl_vulkan_wrap(pl_gpu gpu, const struct pl_vulkan_wrap_params *params)
 {
+    VkFormat desired_fmt = params->format;
+    if (desired_fmt == VK_FORMAT_R10X6_UNORM_PACK16)
+        desired_fmt = VK_FORMAT_R16_UNORM;
+    else if (desired_fmt == VK_FORMAT_R10X6G10X6_UNORM_2PACK16)
+        desired_fmt = VK_FORMAT_R16G16_UNORM;
     pl_fmt fmt = NULL;
     for (int i = 0; i < gpu->num_formats; i++) {
         const struct vk_format **vkfmt = PL_PRIV(gpu->formats[i]);
-        if ((*vkfmt)->tfmt == params->format) {
+        if ((*vkfmt)->tfmt == desired_fmt) {
             fmt = gpu->formats[i];
             break;
         }
@@ -1155,7 +1160,7 @@ pl_tex pl_vulkan_wrap(pl_gpu gpu, const struct pl_vulkan_wrap_params *params)
 
     if (!fmt) {
         PL_ERR(gpu, "Could not find pl_fmt suitable for wrapped image "
-               "with format %s", vk_fmt_name(params->format));
+               "with format %s", vk_fmt_name(desired_fmt));
         return NULL;
     }
 
@@ -1215,7 +1220,7 @@ pl_tex pl_vulkan_wrap(pl_gpu gpu, const struct pl_vulkan_wrap_params *params)
     tex_vk->external_img = true;
     tex_vk->held = !fmt->num_planes;
     tex_vk->img = params->image;
-    tex_vk->img_fmt = params->format;
+    tex_vk->img_fmt = desired_fmt;
     tex_vk->num_planes = fmt->num_planes;
     tex_vk->usage_flags = usage;
     tex_vk->aspect = params->aspect;
